cv32e40s/tests/programs/custom/pma/pma.c

```diff
@@ -13,319 +13,1147 @@
 
-#define  EXCEPTION_INSN_ACCESS_FAULT      1
-#define  EXCEPTION_LOAD_ACCESS_FAULT      5
-#define  EXCEPTION_STOREAMO_ACCESS_FAULT  7
+// External symbols
+extern volatile uint32_t mtvt_table;
 
-/* CFG Memory Layout
+// Message strings for use in assembly printf
+//const volatile char * const volatile asm_printf_msg_template = "Entered handler %0d\n";
+
+// ---------------------------------------------------------------
+// Generic test template:
+// ---------------------------------------------------------------
+// uint32_t <test_name>(uint32_t index, uint8_t report_name){
+//   volatile uint8_t test_fail = 0;
+//   /* Test variable instantiation */
+//
+//   SET_FUNC_INFO
+//
+//   if (report_name) {
+//     cvprintf(V_LOW, "\"%s\"", name);
+//     return 0;
+//   }
+//
+//   /* Insert test code here /*
+//
+//   if (test_fail) {
+//     cvprintf(V_LOW, "\nTest: \"%s\" FAIL!\n", name);
+//     return index + 1;
+//   }
+//   cvprintf(V_LOW, "\nTest: \"%s\" OK!\n", name);
+//   return 0;
+// }
+// ---------------------------------------------------------------
+
+// ---------------------------------------------------------------
+// Test prototypes - should match
+// uint32_t <name>(uint32_t index, uint8_t report_name)
+//
+// Use template below for implementation
+// ---------------------------------------------------------------
+uint32_t dummy(uint32_t index, uint8_t report_name);
+uint32_t exec_only_for_main_regions(uint32_t index, uint8_t report_name);
+uint32_t non_natural_aligned_store_to_io(uint32_t index, uint8_t report_name);
+uint32_t non_natural_aligned_loads_from_io(uint32_t index, uint8_t report_name);
+uint32_t misaligned_fault_nochange_regfile(uint32_t index, uint8_t report_name);
+uint32_t misaligned_border_io_to_mem(uint32_t index, uint8_t report_name);
+uint32_t misaligned_border_mem_to_io(uint32_t index, uint8_t report_name);
+uint32_t misalign_store_fault_no_bus_access_second(uint32_t index, uint8_t report_name);
+uint32_t misalign_store_fault_io_no_bus_access_first(uint32_t index, uint8_t report_name);
+
+// ---------------------------------------------------------------
+// Prototypes for functions that are test specific and
+// perform the debug part of specific tests.
+// ---------------------------------------------------------------
+//void template_function_dbg(void) __attribute__((section(".debugger"), __noinline__));
+
+// ---------------------------------------------------------------
+// Helper functions
+// ---------------------------------------------------------------
+void increment_mepc(volatile uint32_t incr_val);
+void clear_status_csrs(void);
+
+// IRQ related
+uint32_t detect_irq_mode(void);
+void setup_clic(void);
+
+
+// ---------------------------------------------------------------
+// Helper functions
+// ---------------------------------------------------------------
+/*
+ * set_test_status
  *
- * [r0, "main"]
- * 0xFFFF_FFFF
- * 0x1A11_1000         <MEM_ADDR_1>
```

Comment:
The map of how the address space is removed? Is it because the tests no longer work that way or something else?

Reply:
technically the test is rewritten from scratch, so not removed per se - could add it back in as a comment in the test description.
Could also change my code to use this as defines/global constants, I will look into that.